src/reducers-actions/trades.ts

```diff
@@ -137,10 +140,14 @@ function getPendingTrades(tradesByRevertKey: Map<string, Trade[]>): Map<string,
 function applyRevertsToTrades(
   trades: Trade[],
   reverts: TradeReversion[],
-  pendingTrades?: Map<string, Trade[]>,
-): [Trade[], Map<string, Trade[]>] {
+  pendingTrades?: [string, Trade[]][],
+): [Trade[], [string, Trade[]][]] {
+  // Not trivial to serialize to/from JSON a Map of Sets, so we instead rebuild it here on start
```

Comment:
It's not that hard if you really want to
```js
function processMapSet(value) {
    if (value instanceof Set) {
        return {
            values: Array.from(value, processMapSet),
            __JSON_type: 'Set',
        }
    }
    if (value instanceof Map) {
        const container = {__JSON_type: 'Map'}
        const values = []
        value.forEach((val, key) => {
            values.push([key, processMapSet(val)])
        })
        return {
            values,
            __JSON_type: 'Map',
        }
    }
    return value
}

function replacer(key, value) {
    if (value instanceof Set || value instanceof Map) {
        return processMapSet(value)
    } 

    return value
}

function stringifyMapSets(map) {
    return JSON.stringify(map, replacer, 2)
}

function parseMapSets(str) {
    return JSON.parse(str, reviver)
}

function reviver(key, value) {
    if (typeof value === 'object') {
        if (value.__JSON_type === 'Set') {
            return new Set(value.values)
        }
        if (value.__JSON_type === 'Map') {
            return new Map(value.values)
        }
    }
    
    return value
}

let tradesState = {
    '0x1234': {
        trades: [{t:1},{t:2}],
        pendingTrades: new Map([['0xabcd', new Set([{t:1},{t:2}])], ['0xEF', new Set([{t:3},{t:4}])]]),
        lastCheckedBlockNumber: 1,
    },
    '0x5678': {
        trades: [{t:5},{t:6}],
        pendingTrades: new Map([['0xEF12', new Set([{t:5},{t:6}])], ['0x12ef', new Set([{t:7},{t:8}])]]),
        lastCheckedBlockNumber: 2,
    }
}

> stringifyMapSets(tradesState)
"{
  "0x1234": {
    "trades": [
      {
        "t": 1
      },
      {
        "t": 2
      }
    ],
    "pendingTrades": {
      "values": [
        [
          "0xabcd",
          {
            "values": [
              {
                "t": 1
              },
              {
                "t": 2
              }
            ],
            "__JSON_type": "Set"
          }
        ],
        [
          "0xEF",
          {
            "values": [
              {
                "t": 3
              },
              {
                "t": 4
              }
            ],
            "__JSON_type": "Set"
          }
        ]
      ],
      "__JSON_type": "Map"
    },
    "lastCheckedBlockNumber": 1
  },
  "0x5678": {
    "trades": [
      {
        "t": 5
      },
      {
        "t": 6
      }
    ],
    "pendingTrades": {
      "values": [
        [
          "0xEF12",
          {
            "values": [
              {
                "t": 5
              },
              {
                "t": 6
              }
            ],
            "__JSON_type": "Set"
          }
        ],
        [
          "0x12ef",
          {
            "values": [
              {
                "t": 7
              },
              {
                "t": 8
              }
            ],
            "__JSON_type": "Set"
          }
        ]
      ],
      "__JSON_type": "Map"
    },
    "lastCheckedBlockNumber": 2
  }
}"
```

Here's actually a god use case for **localForage**, but not today

Reply:
^ lol

Reply:
> ```            __JSON_type: 'Map',```

This is what I did NOT want to do =P

Anyway, it was a quick fix for the dumb serialization I had before.
I still think it can be improved, not sure the Map of Sets is the way to go. 

As I mentioned in the description, I was still having duplicates in some cases.
Need to think a bit more about this.